### PR TITLE
[SLAM]Add features of save and load relocalization map.

### PR DESCRIPTION
**Pull request by lisa0314**
@DonnaWuDongxia please help to review.

### CODE REVIEW

File: src/slam/slam_async_tasks.cpp [hunk 1]
```diff
@@ -407,3 +407,46 @@ v8::Local<v8::Value> GetOccupancyMapBoundsTask::GetResolved() {
...
+      (result_status.id() >= rs::core::status_no_error) ? Successful : Failed;
+}
+
+SlamPayload<std::string>* SaveRelocalizationMapTask::GetPayload() {
+  return reinterpret_cast<SlamPayload<std::string>*>(AsyncTask::GetPayload());
+}
+
+v8::Local<v8::Value> SaveRelocalizationMapTask::GetResolved() {
+  auto payload = GetPayload();
+  return Nan::New(payload->data().c_str()).ToLocalChecked();
```
Comment (DonnaWuDongxia) [inline src/slam/slam_async_tasks.cpp]: I have made the return promise ad Promise<void> in #146, so we don't need to return the name any more.

File: src/slam/slam_module.cpp [hunk 2]
```diff
@@ -681,6 +681,28 @@ utils::Status SlamModule::LoadOccupancyMap(const std::string& file_name) {
   utils::Status result;
   result.set_id(slam_->load_occupancy_map(file_name));
   if (result.id() < rs::core::status_no_error) {
+    result.set_message("failed to load relocalization map.");
+  }
+
```
Comment (DonnaWuDongxia) [inline src/slam/slam_module.cpp]: Don't need a blank line here.

File: src/slam/slam_module.cpp [hunk 3]
```diff
@@ -681,6 +681,28 @@ utils::Status SlamModule::LoadOccupancyMap(const std::string& file_name) {
...
+}
+
+utils::Status SlamModule::SaveRelocalizationMap(
+    const std::string& file_name) {
+  utils::Status result;
+  result.set_id(slam_->save_relocalization_map(file_name));
+  if (result.id() < rs::core::status_no_error) {
+    result.set_message("failed to save occpuancy map.");
+  }
+
```
Comment (DonnaWuDongxia) [inline src/slam/slam_module.cpp]: Don't need a blank line here.

File: src/slam/slam_module.cpp [hunk 4]
```diff
@@ -681,6 +681,28 @@ utils::Status SlamModule::LoadOccupancyMap(const std::string& file_name) {
...
+  return result;
+}
+
+utils::Status SlamModule::LoadRelocalizationMap(
+    const std::string& file_name) {
+  utils::Status result;
+  result.set_id(slam_->load_relocalization_map(file_name));
+  if (result.id() < rs::core::status_no_error) {
     result.set_message("failed to load occpuancy map.");
```
Comment (DonnaWuDongxia) [inline src/slam/slam_module.cpp]: occupancy map --> relocalization map ?

**Review by DonnaWuDongxia — COMMENTED** [review]
Looks good to me except some trivial points. Please update the PR and I will merge it. I have the permission now.

**Comment by DonnaWuDongxia** [comment]
@lisa0314 another one, please remember to close the related Github issues with "Fix #xxx", and a space behind "[SLAM]" in the commit log.

**Comment by lisa0314** [comment]
@DonnaWuDongxia , all done.

**Comment by DonnaWuDongxia** [comment]
@lisa0314 a "temp" in the commit log.

**Pull request merged**